plugins/inputs/jti_openconfig_telemetry/openconfig_telemetry.go

```diff
@@ -0,0 +1,324 @@
+		}
+		m.grpcClientConn, err = grpc.Dial(m.Server, grpc.WithTransportCredentials(creds))
+	} else {
+		m.grpcClientConn, err = grpc.Dial(m.Server, grpc.WithInsecure())
+	}
+	if err != nil {
+		log.Fatalf("E! Failed to connect: %v", err)
+	}
+
+	if m.Debug {
+		log.Printf("I! Opened a new gRPC session to %s on port %s", grpc_server, grpc_port)
+	}
+
+	// If username, password and clientId are provided, authenticate user before subscribing for data
+	if m.Username != "" && m.Password != "" && m.ClientId != "" {
+		lc := authentication.NewLoginClient(m.grpcClientConn)
+		loginReply, loginErr := lc.LoginCheck(context.Background(), &authentication.LoginRequest{UserName: m.Username, Password: m.Password, ClientId: m.ClientId})
+		if loginErr != nil {
+			log.Fatalf("E! Could not initiate login check: %v", err)
+		}
+
+		// Check if the user is authenticated. Bail if auth error
+		if !loginReply.Result {
+			log.Fatalf("E! Failed to authenticate the user")
+		}
+	}
+
+	c := telemetry.NewOpenConfigTelemetryClient(m.grpcClientConn)
+
+	for _, sensor := range m.Sensors {
+		go func(sensor string, acc telegraf.Accumulator) {
+			spathSplit := strings.SplitN(sensor, " ", -1)
+			var sensorName string
+			var pathlist []*telemetry.Path
+			customReportingRate := false
+			var reportingRate uint64
+
+			reportingRate = uint64(m.SampleFrequency)
+
+			if len(spathSplit) > 1 {
+				for i, path := range spathSplit {
+					// We allow custom reporting rate per sensor by specifying rate in milliseconds
+					// followed by measurement name or sensor path
+					if i == 0 {
+						sensorName = path
+						reportingRate, err = strconv.ParseUint(path, 10, 32)
+						if err == nil {
+							customReportingRate = true
+						} else {
+							reportingRate = uint64(m.SampleFrequency)
+							// If the first word is not an integer and starts with /, we treat it as both sensor and a measurement name
+							if strings.HasPrefix(sensorName, "/") {
+								pathlist = append(pathlist, &telemetry.Path{Path: path, SampleFrequency: uint32(reportingRate)})
+							}
+						}
+					} else if customReportingRate && i == 1 {
+						// If our first word is reporting rate for this list of sensors, we treat second word as measurement name and if it has /
+						// we treat it as another sensor to be monitored
+						sensorName = path
+						if strings.HasPrefix(path, "/") {
+							pathlist = append(pathlist, &telemetry.Path{Path: path, SampleFrequency: uint32(reportingRate)})
+						}
+					} else {
+						pathlist = append(pathlist, &telemetry.Path{Path: path, SampleFrequency: uint32(reportingRate)})
+					}
+				}
+			} else {
+				sensorName = sensor
+				pathlist = append(pathlist, &telemetry.Path{Path: sensor, SampleFrequency: uint32(reportingRate)})
+			}
+			stream, err := c.TelemetrySubscribe(context.Background(),
+				&telemetry.SubscriptionRequest{PathList: pathlist})
+			if err != nil {
+				log.Fatalf("E! Could not subscribe: %v", err)
```

Comment:
Use `acc.AddError`, also should we return from the goroutine here if we can't subscribe.

Reply:
Sure.